### DISCUSSION TITLE
How to use lazy loading with `add` and async?

### DISCUSSION THREAD

**Discussion by qarmin** [sqlalchemy/sqlalchemy#8170] [Usage Questions]
SQLAlchemy 1.4, Python 3.9

In my table I have this relation
```
class Patient(Base):
    head_positioning = relationship("HeadPositioning")
```

In select I can use `selectinload` inside options to disable lazy loading and return head_postions related to selected patients, which works fine.
```
                select(Patient).order_by(Patient.id).options(selectinload(Patient.head_positioning)))
```

But when I use `add` I don't see any method how to choose `selectinload` method to this command
```
    async def create_instance_of_object(self, obj: Patient):
            async with self.session.begin():
                self.session.add(obj) # How to add here lazy load options?
                await self.session.commit()
```
Later when I use `obj` when session ends, then I have error(I already use `expire_on_commit=False`)
```
pydantic is not bound to a Session; lazy load operation of attribute 'head_positioning' cannot proceed
```

I already read https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html and tried to use different lazy load method in relationship, but this doesn't work.

**Comment by zzzeek** [comment]
The `session.add()` method doesn't load objects, it is given an object from the user that's put into the session.   It's also in [pending](https://docs.sqlalchemy.org/en/14/glossary.html#term-pending) so there is nothing to load on it in any case.

Usually then, if you created a `Patient` object here, you'd already have associated `head_positioning` with it, and it would be already present,like  `session.add(Patient(head_positioning=HeadPositioning()))`.   But I guess you might be assigning primary key identifiers directly, like `session.add(Patient(head_positioning_id=12))` to be having this problem.   If you are rolling with primary key identifiers directly then you need to make sure what you need is loaded on the object afterwards.

A way you can do this now is to refresh the object using .get() with populate existing:

```py
await session.get(Patient, (patient_pk, ), populate_existing=True, options=[selectinload(Patient.head_positioning)])
```

that said, if this is a many-to-one, it's likely simpler to just load the `HeadPositioning` object you need directly and just assign it rather than using PK ids.
